byron/chain/executable-spec/src/Cardano/Spec/Chain/STS/Block.hs

```diff
@@ -94,18 +101,27 @@ bUpdPayload b = (b ^. bBody ^. bUpdProp, b ^. bBody ^. bUpdVotes)
 bEndorsment :: Block -> (ProtVer, VKey)
 bEndorsment b = (b ^. bBody ^. bProtVer, b ^. bHeader ^. bhIssuer)
 
-
 -- | Compute the size (in words) that a block takes.
 bSize :: Block -> Natural
-bSize = fromIntegral . abstractSize acctMap
+bSize b = bHeaderSize (b ^. bHeader) + bBodySize (b ^. bBody)
+
+-- | Compute the size (in words) that a block body occupies.
+bBodySize :: BlockBody -> Natural
+bBodySize = fromIntegral . abstractSize costs
   where
-    acctMap = []
+    costs = Map.fromList [ (typeOf (undefined::Maybe UProp), 1)
+                         , (typeOf (undefined::ProtVer), 1)
+                         , (typeOf (undefined::DCert), 1)
+                         , (typeOf (undefined::TxWits TxId), 1)]
 
--- | Compute the size (in words) that a block header.
+-- | Compute the size (in words) that a block header occupies.
 bHeaderSize :: BlockHeader -> Natural
-bHeaderSize = fromIntegral . abstractSize acctMap
+bHeaderSize = fromIntegral . abstractSize costs
```

Comment:
This loos good! Although I'm thinking, if the cost is constant, we might as well define the `bHeaderSize` as `1`. But I'd leave this as is for now and ask @nc6 what does he think.